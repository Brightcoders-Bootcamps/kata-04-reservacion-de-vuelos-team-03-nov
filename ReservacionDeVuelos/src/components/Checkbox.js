import React, {useState} from 'react';
import {StyleSheet, View, Text} from 'react-native';
import colors from '../utils/Colors';
import CheckBox from '@react-native-community/checkbox';
import contentText from '../utils/Constants';

function CheckBoxComponent() {
  const [isSelected, setSelection] = useState(false);
  const [isSelected2, setSelection2] = useState(false);
  return (
    <>
      <View style={styles.container}>
        <View style={styles.checkboxContainer}>
          <CheckBox
            value={isSelected}
            onValueChange={setSelection}
            style={styles.checkbox}
            tintColors={{true: colors.blue, false: colors.black}}
          />
          <Text style={styles.labelCheckBox}>
            I agree to the
            <Text style={{textDecorationLine: 'underline'}}> Terms</Text> and
            <Text style={{textDecorationLine: 'underline'}}>
              {' '}
              Privacy Policy.
            </Text>
            *
          </Text>
        </View>
        <View style={styles.checkboxContainer}>
          <CheckBox
            value={isSelected2}
            onValueChange={setSelection2}
            style={styles.checkbox}
            tintColors={{true: colors.blue, false: colors.black}}
          />
          <Text style={styles.labelCheckBox}> {contentText.subscribe} </Text>
        </View>
      </View>
    </>
  );
}

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 2,
    marginTop: 15,
    marginLeft: -18,
  },
  checkboxContainer: {
    flexDirection: 'row',
    marginBottom: 5,
    width: '90%',
  },
  labelCheckBox: {
    fontSize: 13,
    margin: 8,
    color: colors.darkGray,
  },
  checkbox: {
    alignSelf: 'center',
  },
  label: {
    margin: 8,
  },
});

export default CheckBoxComponent;
