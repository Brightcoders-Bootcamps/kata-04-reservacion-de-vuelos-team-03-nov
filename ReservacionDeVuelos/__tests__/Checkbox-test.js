import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from '../src/components/Checkbox';

test('renders correctly', () => {
  const tree = renderer.create(<Checkbox />).toJSON();
  expect(tree).toMatchSnapshot();
});
