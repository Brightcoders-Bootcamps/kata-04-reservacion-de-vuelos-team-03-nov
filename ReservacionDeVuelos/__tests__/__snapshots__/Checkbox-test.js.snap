// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 2,
      "marginTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 5,
        "width": "90%",
      }
    }
  >
    <View
      pointerEvents="auto"
    >
      <RNCCheckbox
        forwardedRef={null}
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 32,
              "width": 32,
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
        tintColors={
          Object {
            "false": "#000",
            "true": "#046BF1",
          }
        }
        value={false}
      />
    </View>
    <Text
      style={
        Object {
          "color": "#757575",
          "fontSize": 13,
          "margin": 8,
        }
      }
    >
      I agree to the
      <Text
        style={
          Object {
            "textDecorationLine": "underline",
          }
        }
      >
         Terms
      </Text>
       and
      <Text
        style={
          Object {
            "textDecorationLine": "underline",
          }
        }
      >
        Privacy Policy.
      </Text>
      *
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 5,
        "width": "90%",
      }
    }
  >
    <View
      pointerEvents="auto"
    >
      <RNCCheckbox
        forwardedRef={null}
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 32,
              "width": 32,
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
        tintColors={
          Object {
            "false": "#000",
            "true": "#046BF1",
          }
        }
        value={false}
      />
    </View>
    <Text
      style={
        Object {
          "color": "#757575",
          "fontSize": 13,
          "margin": 8,
        }
      }
    >
       
      Subscribe for select product updates.
       
    </Text>
  </View>
</View>
`;
